nodejs/lancedb/index.ts

```diff
@@ -152,18 +156,33 @@ export async function connect(
  */
 export async function connect(
   options: Partial<ConnectionOptions> & { uri: string },
+  session?: Session,
 ): Promise<Connection>;
 export async function connect(
   uriOrOptions: string | (Partial<ConnectionOptions> & { uri: string }),
-  options: Partial<ConnectionOptions> = {},
+  optionsOrSession?: Partial<ConnectionOptions> | Session,
+  session?: Session,
```

Comment:
Could we actually move `session` into `ConnectionOptions`? Then a lot of these changes aren't necessary.